web/docs/language/features.md

```diff
@@ -178,138 +178,249 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
+Wasp's `useQuery` hook behaves mostly the same as [_react-query_'s `useQuery` hook](https://react-query.tanstack.com/docs/api#usequery), the only difference being in not having to supply the key. Wasp handles the caching automatically under the hood.
 
-Example of usage:
-```js
+Here's an example of calling the Queries using the `useQuery` hook:
+```jsx
 import React from 'react'
 import { useQuery } from '@wasp/queries'
-import getTasks from '@wasp/queries/getTasks'
 
-const MyComponent = (props) => {
-  const { data: tasks, error } = useQuery(getTasks)
-  return <div> { JSON.stringify(tasks || error) } </div>
+import fetchAllTasks from '@wasp/queries/fetchAllTasks'
+import fetchFilteredTasks from '@wasp/queries/fetchFilteredTasks'
+
+
+
+const MainPage = () => {
+  const {
+    data: allTasks,
+    error: error1
+  } = useQuery(fetchAllTasks)
+
+  const {
+    data: doneTasks,
+    error: error2
+  } = useQuery(fetchFilteredTasks, { isDone: true})
+
+  return (
+    <div> 
+        <p>All tasks: { JSON.stringify(allTasks || error1) }</p>
+        <p>Finished tasks: { JSON.stringify(doneTasks || error2) }</p>
+    </div>
+  )
 }
+
+export default MainPage
 ```
 
-#### Error handling
-For security reasons, all errors thrown in the query NodeJS function are sent to the client via HTTP API as 500 errors, with any further details removed, so that any unpredicted errors don't make it out with possibly sensitive data.
+#### Error Handling
+For security reasons, all exceptions thrown in the Query's NodeJS implementation are sent to the client as responses with the HTTP status code `500`, with all other details removed.
+Removing error information helps against accidentally leaking possibly sensitive information over the network.
 
-If you do want to throw an error that will pass some information to the client, you can use `HttpError` in your NodeJS query function:
-```js
+If you do want to pass additional error information to the client, you can construct and throw an appropriate `HttpError` in your NodeJS Query function:
+```js title=ext/queries.js
 import HttpError from '@wasp/core/HttpError.js'
 
-export getTasks = async (args, context) => {
+export const getTasksError = async (args, context) => {
   const statusCode = 403
-  const message = 'You can\'t do this!'.
+  const message = 'You can\'t do this!'
   const data = { foo: 'bar' }
   throw new HttpError(statusCode, message, data)
 }
 ```
 
-and then in client it will be thrown as an Error with corresponding `.message` and `.data` fields (if status code is 4xx - otherwise `message` and `data` will not be forwarded to the client, for security reasons).
-
-This ensures that no error will accidentally leak out from the server, potentionally exposing sensitive data or implementation details.
+If the status code is `4xx`, the client will receive a response object with the corresponding `.message` and `.data` fields.
```

Comment:
This will actually get rethrown as an Error on the client -> that was mentioned before but now it is not anymore.